mitodl/micromasters
# Added provider and switch to User.username for JWT tokens

Blocked by #3988

#### What are the relevant tickets?
Fixes #4074 

#### What's this PR do?
This updates how we create the JWT to send `User.username` as well as `provider='micromasters'`. This will allow discussions to associate the JWT with a MM social auth.

#### How should this be manually tested?
Make sure you are running at least the code in https://github.com/mitodl/open-discussions/pull/973 and follow the instructions within to create a `UserSocialAuth` record. Set `OPEN_DISCUSSIONS_REDIRECT_URL` to /complete/micromasters and verify you can successfully authenticate.
